src/connectors/dynamic-widgets/__tests__/connectDynamicWidgets-test.ts

```diff
@@ -177,6 +177,25 @@ describe('connectDynamicWidgets', () => {
           false
         );
       });
+
+      it('fails when transformItems returns anything else than an array', () => {
+        const renderFn = jest.fn();
+        const widgetParams = {
+          transformItems() {
+            return null as any;
```

Comment:
I think this better communicates that the code is wrong TypeScript-wise?

```suggestion
            // @ts-ignore
            return null;
```

Reply:
```suggestion
            // @ts-expect-error
            return null;
```

even better like this

Reply:
it needs to be on the next line then.
Fixed it here: https://github.com/algolia/instantsearch.js/pull/4975/commits/1b7efdc1ef01498ed91867e71b88c07f55b17444